vllm/worker/neuron_model_runner.py

```diff
@@ -76,6 +81,33 @@ def __init__(
         # Lazy initialization.
         self.model: nn.Module  # initialize after load_model.
 
+        # Once NEURON_ON_DEVICE_SAMPLING_DISABLED is set to True, turn off
+        # on-device sampling
+        self._on_device_sampling_disabled = os.getenv(
+            "NEURON_ON_DEVICE_SAMPLING_DISABLED", False) == 'True'
+
+        # NEURON needs to update sampling parameters when request IDs change
+        # across batches. This variable stores the previous batch's request IDs
+        # to determine if an update is needed.
+        self._previous_batch_request_ids: List[str] = []
```

Comment:
did you verify that order of request ids is deterministic?

Reply:
yes! also if there is no new sequence group but the order is changed, we will need to update the config as well, as the on-device sampling parameters are ordered.